text/src/main/java/net/skyscanner/backpack/text/BpkText.kt

```diff
@@ -0,0 +1,60 @@
+package net.skyscanner.backpack.text
+
+import android.content.Context
+import android.content.res.TypedArray
+import android.support.v4.widget.TextViewCompat
+import android.support.v7.widget.AppCompatTextView
+import android.util.AttributeSet
+
+
+open class BpkText(
+  context: Context,
+  attrs: AttributeSet?,
+  defStyleAttr: Int) : AppCompatTextView(context, attrs, defStyleAttr) {
+
+  companion object {
+    val XS = 0
+    val SM = 1
+    val BASE = 2
+    val LG = 3
+    val XL = 4
+    val XXL = 5
+  }
+
+  var textStyle: Int = BASE;
```

Comment:
Since those properties are public and mutable we need to react to updates here (possibly by calling setup again?). e.g: 
```
var textStyle: Int = BASE;
  set(value) {
     field = value
     setup()
  }
```